docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Define-Providers-Interface
+authors:
+  - "@Gal-Zaidman"
+creation-date: 2021-07-25
+last-updated: 2021-08-23
+---
+
+# Define Providers Interface
+
+## Summary
+
+Currently, the assisted installer enables you to install on any platform where you can boot the discovery ISO
+resulting in a cluster with platform set to none, and also supports vSphere and Bare-Metal platforms.
+
+Openshift can also be installed on other on-prem providers such as:
+- Red Hat Virtualization (oVirt).
+- OpenStack.
+
+Currently, the assisted installer supports only the Bare-Metal and vSphere platforms, but for future on-prem
+providers we need to make sure there is a clear interface each on prem provider can implement to add its platform.
+
+This enhancement will define a clear interface to extend the assisted installer to add on prem provider integrations.
+
+## Motivation
+
+Assisted Installer can be a great tool to perform User Provisioned installation (UPI)  installation on external providers,
+providing the user with clear UI to see its provisioned machines joining the cluster, and lowering installation
+resources requirements.
+It is only natural that on-prem providers will want to follow vSphere path and extend assisted installer to support
+their platform, to make sure that the provider specific code is easy to maintain, doesn't burden the assisted installer team,
+and the providers have a clear understanding of what they need to implement to add support to their platform we need to
+define a clear interface which each provider can implement and maintain.
+
+### Goals
+
+- Define a clear way to add a specific platform provider to the assisted installer.
+- Migrate all existing supported providers to implement the same unified interface.
+
+### Non-Goals
+
+- Support IPI installations.
+- Integrating any relevant [Openshift installer types](https://github.com/openshift/installer/tree/master/pkg/types)
+into the assisted installer code - this will require a large refactor of the assisted installer code.
+- Remove Baremetal from not appropriate struct/function names.
+- Adding Support for cloud installations (AWS/GCP/Azure).
```

Comment:
I find this bullet similar to the 2nd non-goal, but a bit confusing since specifying only the cloud installations while there are much more.
IIUC, your intention is this enhancement doesn't imply supporting any OpenShift Installer supported providers, regardless if they are cloud providers or not.

Reply:
done